core/controlplane/config/templates/cloud-config-controller

```diff
@@ -1,3 +1,21 @@
+{{ define "instance" -}}
+{{- $S3URI := self.Parts.s3.Asset.URL -}}
```

Comment:
I like `self.s3.Asset.URL` 👍 

Btw, I'd prefer the `UserData` type to be a struct with fields like `s3` and `instance` rather than a map. For me. map[string] looks a bit confusing - Even though the only expected keys for it are `s3` and `instance`, `map[string]` doesn't tell that to us.